plugin/kubernetes/object/endpoint.go

```diff
@@ -43,8 +45,19 @@ type EndpointPort struct {
 // EndpointsKey return a string using for the index.
 func EndpointsKey(name, namespace string) string { return name + "." + namespace }
 
-// ToEndpoints converts an api.Endpoints to a *Endpoints.
-func ToEndpoints(end *api.Endpoints) *Endpoints {
+// ToEndpoints returns a function that converts an *api.Endpoints to a *Endpoints.
+func ToEndpoints(skipCleanup bool) ToFunc {
```

Comment:
Where does the ToFunc requirement come from? Just testing?

Reply:
This is required to allow the Endpoints watch to use the default processor, instead of the current in-line processor.